[FEATURE] Have gradle build enforce/detect required jdk 8

If you build with a new JDK, you may get errors that don't provide much clue about the underlying cause of the problem. For example, running:

```
./gradlew integrationTest
```
If your default JDK is say: 11, 
```
$ java -version
openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.4+11, mixed mode)
```

you will see errors like this:

```
org.sonatype.gradle.plugins.scan.ScanIT_Gradle_Versions_4_2_to_4_4 > testAuditTask_Vulnerabilities_OssIndex[Version: 4.2.1] FAILED
    java.lang.IllegalStateException
        Caused by: org.gradle.tooling.GradleConnectionException
            Caused by: java.lang.IllegalArgumentException

```

What, you didn't recognize this as a JDK version issue right away? Me neither. ;)

* **How could we solve this issue? (Not knowing is okay!)**

I'm pretty sure there is a way to enforce JDK version in a Gradle build, not sure how though.

cc @bhamail / @DarthHater / @guillermo-varela

--- a/build.gradle
+++ b/build.gradle
@@ -31,7 +31,7 @@ java {
   targetCompatibility = JavaVersion.VERSION_1_8
 }
 
-if (JavaVersion.current() != JavaVersion.VERSION_1_8) {
+if (JavaVersion.current() != targetCompatibility) {
   throw new GradleException(
       "The java version used ${JavaVersion.current()} is not the expected version ${project.targetCompatibility}.")
 }

--- a/build.gradle
+++ b/build.gradle
@@ -31,6 +31,12 @@ java {
   targetCompatibility = JavaVersion.VERSION_1_8
 }
 
+if (JavaVersion.current() != JavaVersion.VERSION_1_8) {
+  throw new GradleException(
+      "The java version used ${JavaVersion.current()} is not the expected version ${project.targetCompatibility}.")
+}
+
+
 jar {
   manifest {
     attributes(

--- a/build.gradle
+++ b/build.gradle
@@ -36,7 +36,6 @@ if (JavaVersion.current() != targetCompatibility) {
       "The java version used ${JavaVersion.current()} is not the expected version ${project.targetCompatibility}.")
 }
 
-
 jar {
   manifest {
     attributes(

